pkg/querier/querier.go

```diff
@@ -122,6 +127,9 @@ func (q *Querier) Query(ctx context.Context, req *logproto.QueryRequest) (*logpr
 }
 
 func (q *Querier) getQueryIterators(ctx context.Context, req *logproto.QueryRequest) ([]iter.EntryIterator, error) {
+	// Enforce the query timeout while querying backends
+	ctx, _ = context.WithDeadline(ctx, time.Now().Add(q.cfg.QueryTimeout))
```

Comment:
This is called from line 115, where you are already passing in a `context.WithDeadline`. Why is a new one required?

Reply:
It was a mistake. Sorry for pointing out.